src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -431,6 +431,11 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
                     Error(diagnostics, ErrorCode.ERR_FieldCantBeRefAny, argumentSyntax, argumentType);
                 }
 
+                if (argumentType.IsPointerType())
+                {
+                    Error(diagnostics, ErrorCode.ERR_BadTypeArgument, argumentSyntax, argumentType);
```

Comment:
@jcouv 

I should do the same in `Binder_Expressions.cs` as well (check for pointers and produce CS0306?